magicctapipe/image/calib.py

```diff
@@ -0,0 +1,134 @@
+import numpy as np
+from ctapipe.image import (
+    apply_time_delta_cleaning,   
+    number_of_islands,
+    tailcuts_clean,   
+)
+from ctapipe.instrument import CameraGeometry
+from lstchain.image.cleaning import apply_dynamic_cleaning
+from lstchain.image.modifier import (
+    add_noise_in_pixels,
+    random_psf_smearer,  
+    set_numba_seed  
+)
+from magicctapipe.image import MAGICClean
+
+
+__all__ = [
+    "calibrate"
+]
+
+
+def calibrate(event, tel_id, config, calibrator, LST_bool, obs_id=None, camera_geoms=None, magic_clean=None):
+
+    """
+    This function calibrates the camera image for a single event of a telescope 
+
+    Parameters
+    ----------
+    event: event 
+        From an EventSource
+    tel_id: int
+        Telescope ID     
+    config: dictionary
+        Parameters for image extraction and calibration    
+    calibrator: CameraCalibrator (ctapipe.calib)
+        ctapipe object needed to calibrate the camera
+    LST_bool: bool
+        Whether the telescope is a LST
+    obs_id: int
+        Observation ID. Unsed in case of LST telescope
+    camera_geoms: telescope.camera.geometry
+        Camera geometry. Used in case of LST telescope
+    magic_clean: dictionary (1 entry per MAGIC telescope)
+        Each entry is a MAGICClean object using the telescope camera geometry. Used in case of MAGIC telescope
+    
+
+    Returns
+    -------
+    signal_pixels: boolean mask
+        Mask of the pixels selected by the cleaning
+    image: numpy array
+        Array of number of p.e. in the camera pixels
+    peak_time: numpy array
+        Array of the signal peak time in the camera pixels
+
+    """
+    
+    calibrator._calibrate_dl0(event, tel_id)
+    calibrator._calibrate_dl1(event, tel_id)
+
+    image = event.dl1.tel[tel_id].image.astype(np.float64)
+    peak_time = event.dl1.tel[tel_id].peak_time.astype(np.float64)
+    
+    if (LST_bool==False) and (magic_clean!=None):
+        if type(magic_clean[tel_id])==MAGICClean: 
```

Comment:
A specific example of what I wrote in the comment below, this code could substitute lines 65-76:

```
        if type(magic_clean[tel_id]) != MAGICClean: 
            raise ValueError("Check the provided magic_clean parameter; calibration was not possible")

        use_charge_correction = config["charge_correction"]["use"]

        if use_charge_correction:
            # Scale the charges by the correction factor
            image *= config["charge_correction"]["factor"]
        # Apply the image cleaning
        signal_pixels, image, peak_time = magic_clean[tel_id].clean_image(
            event_image=image, event_pulse_time=peak_time
        )
```

this has one line less and also the bulk of the code has one layer of indent less, which makes it easier to read
same can be done with the exceptions thrown later on in the code

Reply:
ups, sorry, it seems that somehow I checked this part in an outdated code, closing now